### ISSUE TITLE
Contributor key not found

### ISSUE THREAD

**Issue by daodesigner** [clrfund/monorepo#420]
When trying to reallocate votes from the same machine/browser after original cast user received the following error:
![image](https://user-images.githubusercontent.com/58278045/147488965-22f57c12-8804-4c2f-96c5-50848ce1ee19.jpeg)
![image](https://user-images.githubusercontent.com/58278045/147489005-0a37c673-f8f0-4056-a022-b8e66e17f388.jpeg)

Also occurs if user didn't sign the last tx to send the confirmation on metamask and refreshes the page.

@pettinarip @samajammin you guys seeing this on your end as well?

**Comment by pettinarip** [comment]
I haven't seen this before but looking into the code, this error can only appear if we couldn't load the contributor data from the GUN storage. Wondering if the user has some kind of local storage blocker that blocks us from using GUN.

Will take a closer look tomorrow to see if I can replicate it.

**Comment by pettinarip** [comment]
@daodesigner I could reproduce the issue locally. What I did is to make the contribution and then clear the local storage. If you refresh the page you will see that error.

I think this is an issue we have to live with for now, that occurs if the user wipes his/her storage manually. This is how it was designed I think, not sure if we can modify it and store the contributor and contribution state in a different place (e.g. ipfs).

@samajammin perhaps you have more context on this.

**Comment by daodesigner** [comment]
@crisgarner

**Comment by samajammin** [comment]
Yes we've seen this before - I believe it's the same issue as https://github.com/ethereum/clrfund/issues/129

**Comment by pettinarip** [comment]
I think this issue is related to Gun and how we are configuring it.

As I understand, when we need to store something in Gun, it will store it in your local storage and try to spread it in the network.
- If at that moment, no other peer is connected, the data is only going to live in the client's browser.
- That is why the Relay Peers exists, they have other storing mechanisms and are always available to store things.

Another point is that local storage capacity differs for each browser. I think the avg is 5mb. Perhaps we could also be reaching this limit 🤔 (not sure how we could know that).

My idea around this is to optimize how much data we store in Gun/local storage. Currently, we are storing:
  - Contributor data (key)
  - Cart state
  - Committed cart state

In the cart state, we are storing all the information (i.e. description, title, etc), that is not needed. We could store only the project's indexes and contribution amounts for each. The rest of the information is fetched from the subgraph.
